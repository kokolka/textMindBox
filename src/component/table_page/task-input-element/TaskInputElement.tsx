import React from 'react';
import {
    Formik,
    Form,
    Field
} from 'formik';

interface MyFormValues {
    newTask: string;
}

export const TaskInputElement: React.FC<any> = () => {
    const initialValues: MyFormValues = { newTask: '' };
    return (
        <div>
            <h1>fff</h1>
            <Formik
                initialValues={initialValues}
                onSubmit={(values, actions) => {
                    console.log({ values, actions });
                    alert(JSON.stringify(values, null, 2));
                    actions.setSubmitting(false);
                }}
            >
                
                <Form>
                    <Field id="newTask" name="newTask" placeholder="Введите новое задание" />
                    <button type="submit">Добавить</button>
                </Form>
            </Formik>
        </div>
    )
}

// export default TaskInputElement;
