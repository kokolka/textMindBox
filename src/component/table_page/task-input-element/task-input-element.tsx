import React from 'react';
import {
    Formik,
    FormikHelpers,
    FormikProps,
    Form,
    Field,
    FieldProps,
} from 'formik';

interface MyFormValues {
    newTask: string;
}

const TaskInputElement: React.FC = () => {
    const initialValues: MyFormValues = { newTask: '' };
    return (
        <div>
            <Formik
                initialValues={initialValues}
                onSubmit={(values, actions) => {
                    console.log({ values, actions });
                    alert(JSON.stringify(values, null, 2));
                    actions.setSubmitting(false);
                }}
            >
                <Form>
                    <Field id="newTask" name="newTask" placeholder="Введите новое задание" />
                    <button type="submit">Добавить</button>
                </Form>
            </Formik>
        </div>
    )
}

export default TaskInputElement;
