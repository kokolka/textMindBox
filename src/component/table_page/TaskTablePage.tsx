import React from 'react';
import TaskInputElement from './task-input-element/task-input-element';
import TaskTableElement from './task-table-element/TaskTableElement';


const TaskTablePage: React.FC = () => {
    return(
        <div >
            <TaskInputElement />
            <TaskTableElement s='g'/>
        </div>
    )
};

export default TaskTablePage;