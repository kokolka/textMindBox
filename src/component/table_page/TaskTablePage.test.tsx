import { screen, render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import React from 'react'
import {TaskTablePage} from './TaskTablePage';
import { TaskInputElement } from "./task-input-element/TaskInputElement";

//создание мока внутреннего компонента
jest.mock('./task-input-element/TaskInputElement', () => ({
    TaskInputElement: () => {
        return <div data-testid="TaskInputElement"/>
    }
}));



describe('TaskTablePage', () => {
    it('проверка рендера TaskTablePage', () => {

        const test = renderer.create(<TaskTablePage/>).toJSON();
        expect(test).toMatchSnapshot();
    })
    it('проверка рендера TaskInputElement', () => {

        render(<TaskTablePage/>);
        expect(screen.queryByTestId('TaskInputElement')).toBeInTheDocument();
    })
})
