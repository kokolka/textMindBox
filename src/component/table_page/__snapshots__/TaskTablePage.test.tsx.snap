// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTablePage проверка рендера TaskTablePage 1`] = `
<div>
  <div
    data-testid="TaskInputElement"
  />
</div>
`;
